### PR TITLE
Migrate pro sponsor reactions from icarus4

### PR DESCRIPTION
**Pull request by BobbyTheCatfish**

### CODE REVIEW

File: modules/01-events.js [hunk 1]
```diff
@@ -185,6 +185,20 @@ const Module = new Augur.Module()
...
     }
   } catch (error) { u.errorHandler(error, `User Update Error: ${u.escapeText(newUser?.username)} (${newUser.id})`); }
+})
+.addEvent("messageCreate", async (msg) => {
+  if (!msg.author.bot && msg.guild && msg.guild.id == sf.ldsg) {
+    let i = 0;
+    do {
+      const [sponsor, emoji] = emojis[i];
+      // Sponsor Pings
+      console.log(!(!isNaN(sponsor) && sponsor.length == 18) && msg.content.toLowerCase().includes(sponsor));
```
Comment (LaughLax) [inline modules/01-events.js:195]: This will spam console logs and should be removed.

File: modules/01-events.js [hunk 2]
```diff
@@ -185,6 +185,20 @@ const Module = new Augur.Module()
...
+      if (msg.mentions.members.has(sponsor)) await msg.react(emoji).catch(u.noop);
+      // Filter out any IDs, just in case someone puts in the ID without pinging the person, then do trigger word reactions
+      else if (!(!isNaN(sponsor) && sponsor.length == 18) && Math.random() < 0.3 && msg.content.toLowerCase().includes(sponsor)) await msg.react(emoji).catch(u.noop);
```
Comment (LaughLax) [inline modules/01-events.js:198]: Snowflakes are not guaranteed to be 18 characters long.

File: modules/01-events.js [hunk 3]
```diff
@@ -185,6 +185,20 @@ const Module = new Augur.Module()
       ldsg.channels.cache.get(sf.channels.userupdates).send({ content: `${newUser}: ${newUser.id}`, embeds: [embed] });
     }
   } catch (error) { u.errorHandler(error, `User Update Error: ${u.escapeText(newUser?.username)} (${newUser.id})`); }
+})
+.addEvent("messageCreate", async (msg) => {
+  if (!msg.author.bot && msg.guild && msg.guild.id == sf.ldsg) {
+    let i = 0;
+    do {
```
Comment (LaughLax) [inline modules/01-events.js:192]: Why did you change this to a do-while loop? A for loop is better suited for iterating through an array (both more compact and more readable)

**Review by LaughLax — CHANGES_REQUESTED** [review]
Because the list of Pro Sponsors can change at any time, the snowflakes for the users and their emojis should be kept in the Google Sheet with other server configs rather than in a static file. You can look in `modules/voice.js` for an example of how to load and read that config.

File: modules/01-events.js (hunk 1 above)
Comment (BobbyTheCatfish) [reply modules/01-events.js:195]: Ah, forgot to remove that

File: modules/01-events.js (hunk 3 above)
Comment (BobbyTheCatfish) [reply modules/01-events.js:192]: I thought it would work better with awaiting the reactions, but since it's not exactly necessary I can change it back

Comment (BobbyTheCatfish) [reply modules/01-events.js:192]: On second thought, the way to resolve the snowflake length issue would require trying to fetch a member, which would need to be asynchronous.

File: modules/01-events.js [hunk 4]
```diff
@@ -186,19 +188,29 @@ const Module = new Augur.Module()
...
-      // Sponsor Pings
-      console.log(!(!isNaN(sponsor) && sponsor.length == 18) && msg.content.toLowerCase().includes(sponsor));
+    for (const [sponsor, emoji] of emojis) {
       if (msg.mentions.members.has(sponsor)) await msg.react(emoji).catch(u.noop);
-      // Filter out any IDs, just in case someone puts in the ID without pinging the person, then do trigger word reactions
-      else if (!(!isNaN(sponsor) && sponsor.length == 18) && Math.random() < 0.3 && msg.content.toLowerCase().includes(sponsor)) await msg.react(emoji).catch(u.noop);
-      i++;
-    } while (i < emojis.length);
+      // Filter out sponsors and test for trigger words
+      else if (!await msg.client.users.fetch(sponsor).catch(u.noop) && Math.random() < 0.3 && msg.content.toLowerCase().includes(sponsor)) await msg.react(emoji).catch(u.noop);
```
Comment (LaughLax) [inline modules/01-events.js:210]: We don't have to do a fetch here; checking the guild cache will be sufficient.
```suggestion
      else if (!msg.guild.members.cache.has(sponsor) && Math.random() < 0.3 && msg.content.toLowerCase().includes(sponsor)) await msg.react(emoji).catch(u.noop);
```

File: modules/01-events.js [hunk 5]
```diff
@@ -186,19 +188,29 @@ const Module = new Augur.Module()
...
+.setInit(async () => {
+  const doc = new GoogleSpreadsheet(config.google.sheets.config);
+  try {
+    await doc.useServiceAccountAuth(config.google.creds);
+    await doc.loadInfo();
+    const channels = await doc.sheetsByTitle["Sponsor Channels"].getRows();
+    emojis = Array.from(channels.map(x => [x["Sponsor ID"], x["Sponsor Emoji"]]));
+    emojis = emojis.concat([
+      ["buttermelon", "305039588014161921"],
+      ["noice", "633500960483704843"],
```
Comment (LaughLax) [inline modules/01-events.js:200]: Let's move these snowflakes into `config/snowflakes-example.json` as `emoji.buttermelon` and `emoji.noice`.

**Review by LaughLax — APPROVED** [review]
Good work!
